internal/component/prometheus/exporter/oracledb/oracledb.go

```diff
@@ -48,6 +47,7 @@ type Arguments struct {
 	MaxIdleConns     int               `alloy:"max_idle_conns,attr,optional"`
 	MaxOpenConns     int               `alloy:"max_open_conns,attr,optional"`
 	QueryTimeout     int               `alloy:"query_timeout,attr,optional"`
+	CustomMetrics      alloytypes.OptionalSecret `alloy:"custom_metrics,attr,optional"`
```

Comment:
Second question is: would you use the inline YAML string in .alloy config file? 
Perhaps it's not very usable and it's better to do what postgres exporter does and make it a path to a local YAML file with custom metrics. [See example](https://grafana.com/docs/alloy/next/reference/components/prometheus/prometheus.exporter.postgres/#collect-custom-metrics-from-an-allowlisted-set-of-databases).

Reply:
Hey @thampiotr 
Thanks for the review! I think you are right, i made the tests surrounding this to be in-line string with `local.file.content` , but i think this logic would still work as is with a file path. IE: if we just used [local.file](https://grafana.com/docs/agent/latest/flow/reference/components/local.file/) without content. 

i believe mssql component does the same, whats your thought on this ? Can see here guidance on local.file for 
[query_config](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.mssql/#arguments)

So in this case id modify to string type and change tests to just look for basic filepath string  for compatibility like 
```
local.file "test" {
  filename = "oracle-metrics.yml"
}

prometheus.exporter.oracledb "test" {
    connection_string = CONNECTION_STRING
    custom_metrics = local.file.test 
}
```
with local.file allowing a user to store custom metrics as config or even in source of truth code mgmt repo vs needing to store on disk 

would appreciate your thoughts, and ill make the changes required 
thanks again!

Reply:
The tests look good to me, I don't think they need further changes to use `local.file`.

What I was wondering is whether we should use file path or inline string. I see that your current implementation provides an inline string and I understand that you can use `local.file` to provide it. It does match the `prometheus.exporter.mssql`, but it doesn't match the `prometheus.exporter.postgres` ([example](https://grafana.com/docs/alloy/next/reference/components/prometheus/prometheus.exporter.postgres/#collect-custom-metrics-from-an-allowlisted-set-of-databases)) where the string is provided via file path.

I think that we have more instances where we have an inline string and this approach is a bit more flexible, although more complex. I think we can keep this as-is (pass string inline) and if we wanted to make it easier for users in the future, we could have additional argument of `custom_metrics_path` to reduce pipeline complexity. Or maybe we should have a standard library function that will just load file contents to make it even simpler.

Either way, I think your implementation is fine as-is now and there's no need to further complicate the tests with `local.file`.

Reply:
Makes sense i appreciate this insight -- that said for now if we keep the local.file method and in-line string capability, are you good with leaving this as type `alloytypes.OptionalSecret`  i dont expect users would be embedding secrets into their queries but im curious your take if you still think we should make it to `string` ?

Reply:
@thampiotr just a humble bump when time presents itself - appreciate the help here again ! :)

Reply:
I don't think it's a problem, so overall this LGTM.